### ISSUE TITLE
bug(linux): Surrogate pairs don't work properly in LibreOffice and gnome-terminal

### ISSUE THREAD

**Issue by ermshiperete** [keymanapp/keyman#4429]
**Describe the bug**

This is similar to #4196 but is happening on Linux with LibreOffice.

**To Reproduce**

1. Install the keyboard from [grantha_unicode_phonetic.kmp.zip](https://github.com/keymanapp/keyman/files/5934546/grantha_unicode_phonetic.kmp.zip) (which comes from #4196)
2. open LibreOffice Writer, and with the grantha_unicode_phonetic keyboard enabled, type i+i, p+i, or p+o.
3. compare that with typing in gedit.

Typing in gnome-terminal shows same behavior as LO.

**Expected behavior**
i+i: 𑌈
p+i: 𑌪𑌿
p+o: 𑌪𑍋

**Screenshots**
![Screenshot from 2021-02-05 19-25-30](https://user-images.githubusercontent.com/181336/107075241-22df4d80-67ea-11eb-9e3b-8e1b8a6bc998.png)

**Keyman for Windows/macOS/Linux:**
 - OS: Ubuntu 20.04
 - Keyman version: libkmnkbp0-0: 14.0.236-1+focal1
 - Target Application: LibreOffice 6.4.6.2

**Comment by ermshiperete** [comment]
Things work in gnome-terminal, it's just a rendering problem - see [Ubuntu bug #1916493](https://bugs.launchpad.net/ubuntu/+source/gnome-terminal/+bug/1916493).

**Comment by ermshiperete** [comment]
In **LibreOffice** we end up we an extra '?' (U+003F) in the middle between U+1132A and U+1134B (𑌪 ? 𑍋):

```
F0 91 8C AA | 3F | F0 91 8D 8B
```

(LibreOffice supports surrounding text)

In **Firefox**, which doesn't support surrounding text we end up a with a single U+1134D (𑌪𑍍):

```
F0 91 8D 8D
```

The proper text which we see in **gedit** is U+1132A U+1134B (𑌪𑍋):

```
F0 91 8C AA | F0 91 8D 8B
```

**Comment by ermshiperete** [comment]
The Firefox part seems to be the same reason as #1489.

**Comment by ermshiperete** [comment]
Fixed by reordering patch #7079
